tracing/tempo/http_client.go

```diff
@@ -50,13 +61,24 @@ func NewOtelClient(client http.Client, baseURL *url.URL) (otelClient *OtelHTTPCl
 		}
 	}
 
-	return &OtelHTTPClient{ClusterTag: tags}, nil
+	otelHTTPClient := &OtelHTTPClient{ClusterTag: tags}
+	if config.Get().ExternalServices.Tracing.TempoConfig.CacheEnabled {
+		vanillaStore := store.New[string, *model.TracingSingleTrace]()
```

Comment:
I gave you a bad example earlier. Probably should call this something other than `vanilla` :). Even just `s`  I think is fine since you are using it immediately after. You also should only need to give the type params once here since they can be inferred on the rest of the store.New... methods:
```
s := store.New[string, *model.TracingSingleTrace]()
fifoStore := store.NewFIFOStore(s, config.Get().ExternalServices.Tracing.TempoConfig.CacheCapacity)
otelHTTPClient.TempoCache = store.NewExpirationStore((context.Background(), fifoStore, util.AsPtr(TTL), util.AsPtr(expirationCheckInterval))
```

Reply:
For some reason I have a warning in my IDE, that's why I added it: 
![image](https://github.com/user-attachments/assets/5a996757-0b7e-4b3a-b5c8-c4a0a325c627)